### PR TITLE
182 read wmo mhalo scen

### LINKED ISSUE #182: Cannot read WMO_MHALO.SCEN7
**Describe the bug**

An exception is raised when reading WMO_MHALO.SCEN7 due to not all of the variables being correctly converted to the OpenSCM format, namely 'CH3Br' and a few others.

```
from pymagicc import MAGICC7, MAGICCData
from os.path import join

with MAGICC7() as m:
    d = MAGICCData()
    d.read(join(m.run_dir, 'WMO_MHALO.SCEN7'))
```
The last line raises an exception:
```
Traceback (most recent call last):
  File "/home/jared/.PyCharm2018.3/config/scratches/scratch_10.py", line 7, in <module>
    d.read(join(m.run_dir, 'WMO_MHALO.SCEN7'))
  File "/home/jared/code/uom/magicc/venv/lib/python3.6/site-packages/pymagicc/io.py", line 1700, in read
    self.metadata, self.df = self._read_and_return_metadata_df(filepath)
  File "/home/jared/code/uom/magicc/venv/lib/python3.6/site-packages/pymagicc/io.py", line 1704, in _read_and_return_metadata_df
    return reader.read()
  File "/home/jared/code/uom/magicc/venv/lib/python3.6/site-packages/pymagicc/io.py", line 136, in read
    df, metadata = self.process_data(stream, metadata)
  File "/home/jared/code/uom/magicc/venv/lib/python3.6/site-packages/pymagicc/io.py", line 212, in process_data
    ch, metadata = self._get_column_headers_and_update_metadata(stream, metadata)
  File "/home/jared/code/uom/magicc/venv/lib/python3.6/site-packages/pymagicc/io.py", line 544, in _get_column_headers_and_update_metadata
    column_headers = self._read_units(column_headers)
  File "/home/jared/code/uom/magicc/venv/lib/python3.6/site-packages/pymagicc/io.py", line 504, in _read_units
    emissions_unit = variable.split(DATA_HIERARCHY_SEPARATOR)[1]
IndexError: list index out of range
```

**Expected behavior**

All SCEN7 files can be successfully read. If a variable can't be converted it shouldn't crash

### PR DESCRIPTION
**Pull request by znicholls**
# Pull request

Please confirm that this pull request has done the following:

- [x] Tests added
- [x] Documentation added (where applicable)
- [x] Example added (either to an existing notebook or as a new notebook, where applicable)
- [x] Description in ``CHANGELOG.rst`` added

## Adding to CHANGELOG.rst

Please add a single line in the changelog notes similar to one of the following:

```
- (`#XX <http://link-to-pr.com>`_) Added feature which does something
- (`#XX <http://link-to-pr.com>`_) Fixed bug identified in (`#XX <http://link-to-issue.com>`_)
```

closes #182

### CODE REVIEW

**Comment by lewisjared** [comment]
I also manually reran the Travis run for py3.5 as it had a non-pymagicc error

**Comment by znicholls** [comment]
> It would be nice to add a test for the edge case for `apply_string_substitutions` where the substitutions dict contains two keys which are the same, but have different cases (i.e. butter and Butter) when case_insensitive is True. That should probably raise a ValueError - might be a good safety net as the substitution array is dynamically generated

Done. Take a look then hit the button when you're ready?

**Comment by rgieseke** [comment]
Why does this have "182 read wmo mhalo scen #183" two numbers? ;-)

**Comment by znicholls** [comment]
Cause I mucked up naming. It's meant to be merge request 183, fixes issue
182

File: tests/test_utils.py [hunk 1]
```diff
@@ -127,3 +127,36 @@ def test_apply_string_substitutions_unused_substitutions(
...
+        ("teeth", {"teeth": "tooth", "Teeth": "tooth"}, "tooth"),
+    ],
+)
+def test_apply_string_substitutions_duplicate_substitutions_case_insensitive(
+    inputs, substitutions, expected
+):
+    res = apply_string_substitutions(inputs, substitutions)
+    assert res == expected
+
+    error_msg = "Duplicate substitutions for keys teeth"
```
Comment (rgieseke) [inline tests/test_utils.py]: I think this should not contain the key explicitly, they could be changed, right?

Comment (rgieseke) [reply tests/test_utils.py]: Ah, outdated commit, sorry :-)
